### PR TITLE
fix: [APPAI-1819] Simplified go manifest generation command

### PR DESCRIPTION
**Pull request by dgpatelgit**
Simplified go manifest command executed during SA. Address issue raised by a user https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension/issues/504

### CODE REVIEW

File: src/ProjectDataProvider.ts [hunk 1]
```diff
@@ -325,11 +325,6 @@ export module ProjectDataProvider {
       }
 
       const cmd: string = [
-        Config.getGoExecutable(),
-        `get`,
```
Comment (Ortega-Dan) [inline src/ProjectDataProvider.ts:329]: The problem seems to be with the go run itself and the issue shows how it is not working.

**Review by Ortega-Dan — COMMENTED** [review]
The problem continues with this PR, and the new problem is shown in the issue. (#504)

File: src/ProjectDataProvider.ts (hunk 1 above)
Comment (dgpatelgit) [reply src/ProjectDataProvider.ts:329]: I have some improvements by setting GOPATH to temporary directory. I will share extension for you to verify it once.

File: src/ProjectDataProvider.ts [hunk 2]
```diff
@@ -319,23 +319,35 @@ export module ProjectDataProvider {
...
         fs.mkdirSync(targetDir);
       }
 
       const cmd: string = [
+        `export`,
+        `CRDA_GOPATH=$GOPATH`,
+        `&&`,
+        `export`,
+        `GOPATH=${targetDir}`,
+        `&&`,
```
Comment (arajkumar) [inline src/ProjectDataProvider.ts:334]: This is incorrect and will not work in Windows. If you want to set an env variable, use `env` option.

e.g.

```
   exec(
        cmd,
        { maxBuffer: 1024 * 1200, env: Object.assign(process.env, {"GOPATH": `"${targetDir}"`}) }
....
```

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: src/ProjectDataProvider.ts (hunk 2 above)
Comment (dgpatelgit) [reply src/ProjectDataProvider.ts]: @arajkumar  Now getting temporary path from OS and setting that as GOPATH to get `gomanifest` module. Also GOPATH is set in ENV and passed to exec command. PTAL

File: src/ProjectDataProvider.ts [hunk 3]
```diff
@@ -330,19 +334,20 @@ export module ProjectDataProvider {
...
         `"${goGraphFilePath}"`,
-        `"${Config.getGoExecutable()}"`
+        `"${Config.getGoExecutable()}"`,
       ].join(' ');
 
       console.log('CMD : ' + cmd);
       outputChannelDep.addMsgOutputChannel('\n CMD :' + cmd);
+      let env = process.env
+      env.GOPATH = goPath;
+      outputChannelDep.addMsgOutputChannel('\n env.GOPATH : ' + env.GOPATH);
```
Comment (arajkumar) [inline src/ProjectDataProvider.ts:347]: This is incorrect, you are mutating the global process.env. Previously suggested snippet should do the job.

Comment (dgpatelgit) [reply src/ProjectDataProvider.ts]: My bad, changes are taken in PR.

File: src/ProjectDataProvider.ts [hunk 4]
```diff
@@ -320,6 +320,10 @@ export module ProjectDataProvider {
       let targetDir = paths.join(vscodeRootpath, 'target');
       const goGraphFilePath = paths.join(targetDir, 'golist.json');
 
+      const goPath = paths.join(os.tmpdir(), 'gomanifest');
+      const goManifestPath = paths.join(paths.join(goPath, 'bin'), 'gomanifest');
+
+
```
Comment (arajkumar) [inline src/ProjectDataProvider.ts:326]: nit
```suggestion
```

**Review by arajkumar — APPROVED** [review]
lgtm

**Pull request merged**